src/tss2-fapi/api/Fapi_Provision.c

```diff
@@ -957,6 +960,53 @@ Fapi_Provision_Finish(FAPI_CONTEXT *context)
             return_try_again(r);
             goto_if_error(r, "Init primary finish.", error_cleanup);
 
+            if (command->public_templ.persistent_handle & command->srk_exists) {
+
+                /* It has to be checked whether the public data of the existing persistent
+                   SRK is equal to the public data of the generated key. */
+                r = Esys_TR_FromTPMPublic_Async(context->esys,
+                                                command->public_templ.persistent_handle,
+                                                ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE);
+                goto_if_error(r, "Read public async", error_cleanup);
+
+            } else {
+                context->state = PROVISION_CHECK_SRK_EVICT_CONTROL;
+                return TSS2_FAPI_RC_TRY_AGAIN;
+            }
+            fallthrough;
+
+        statecase(context->state, PROVISION_SRK_GET_PERSISTENT_NAME);
+
+           /* Determine the name of the generated key. */
+           r = Esys_TR_GetName(context->esys, context->srk_handle, &srk_name);
```

Comment:
There must not be other calls to Esys between _Async and _Finish.
So this needs to be moved down a few lines.